src/sdk/namespace_fs.js

```diff
@@ -1564,36 +1564,35 @@ class NamespaceFS {
     // Can be finetuned further on if needed and inserting the Semaphore logic inside
     // Instead of wrapping the whole _upload_stream function (q_buffers lives outside of the data scope of the stream)
     async _upload_stream({ fs_context, params, target_file, object_sdk, offset }) {
-        const { source_stream, copy_source } = params;
+        const { copy_source } = params;
         try {
-            // Not using async iterators with ReadableStreams due to unsettled promises issues on abort/destroy
-            const md5_enabled = this._is_force_md5_enabled(object_sdk);
-            const chunk_fs = new ChunkFS({
+            const md5_enabled =
+                config.NSFS_CALCULATE_MD5 ||
+                this.force_md5_etag ||
+                object_sdk?.requesting_account?.force_md5_etag;
```

Comment:
better use _is_force_md5_enabled() - it probably didn't exist when I wrote this code